tests-integration/tests/splitter.rs

```diff
@@ -99,3 +97,209 @@ fn test_splitter() {
+            ("recipient1", vec![]),
+            ("recipient2", vec![]),
+        ])
+        .with_contracts(vec![
+            ("app-contract", mock_andromeda_app()),
+            ("splitter", mock_andromeda_splitter()),
+            ("cw20", mock_andromeda_cw20()),
+        ])
+        .build(&mut router);
+    let owner = andr.get_wallet("owner");
+    let recipient_1 = andr.get_wallet("recipient1");
+    let recipient_2 = andr.get_wallet("recipient2");
+
+    let app_code_id = andr.get_code_id(&mut router, "app-contract");
+
+    let initial_balances = vec![Cw20Coin {
+        address: owner.to_string(),
+        amount: Uint128::new(1000000u128),
+    }];
+
+    let cw20_init_msg = mock_cw20_instantiate_msg(
+        None,
+        "Test Tokens".to_string(),
+        "TTT".to_string(),
+        6,
+        initial_balances.clone(),
+        Some(mock_minter(
+            owner.to_string(),
+            Some(Uint128::from(10000000000u128)),
+        )),
+        andr.kernel.addr().to_string(),
+    );
+    let cw20_component = AppComponent::new(
+        "cw20".to_string(),
+        "cw20".to_string(),
+        to_json_binary(&cw20_init_msg).unwrap(),
+    );
+
+    let splitter_recipients = vec![
+        AddressPercent {
+            recipient: Recipient::from_string(recipient_1.to_string()),
+            percent: Decimal::from_str("0.2").unwrap(),
+        },
+        AddressPercent {
+            recipient: Recipient::from_string(recipient_2.to_string()),
+            percent: Decimal::from_str("0.8").unwrap(),
+        },
+    ];
+
+    let splitter_init_msg = mock_splitter_instantiate_msg(
+        splitter_recipients,
+        andr.kernel.addr().clone(),
+        None,
+        None,
+        None,
+    );
+    let splitter_app_component = AppComponent {
+        name: "splitter".to_string(),
+        component_type: ComponentType::new(splitter_init_msg),
+        ado_type: "splitter".to_string(),
+    };
+
+    let app_components = vec![splitter_app_component.clone(), cw20_component.clone()];
+    let app = MockAppContract::instantiate(
+        app_code_id,
+        owner,
+        &mut router,
+        "Splitter App",
+        app_components,
+        andr.kernel.addr(),
+        None,
+    );
+
+    let splitter: MockSplitter =
+        app.query_ado_by_component_name(&router, splitter_app_component.name);
+
+    let cw20: MockCW20 = app.query_ado_by_component_name(&router, cw20_component.name);
+
+    let hook_msg = Cw20HookMsg::Send { config: None };
+
+    cw20.execute_send(
+        &mut router,
+        owner.clone(),
+        splitter.addr(),
+        Uint128::new(10),
+        &hook_msg,
+    )
+    .unwrap();
+
+    let cw20_balance = cw20.query_balance(&router, recipient_1);
+    assert_eq!(cw20_balance, Uint128::from(2u128));
+    let cw20_balance = cw20.query_balance(&router, recipient_2);
+    assert_eq!(cw20_balance, Uint128::from(8u128));
+}
+
+#[test]
+fn test_splitter_cw20_with_remainder() {
```

Comment:
Nit: I think we can merge this test with the above and use something like `rstest` to create test cases. This would make it easier to add other potential checks quickly.